feat: initial multi-proof test

LHerskind opened the pull request:
Work in progress for multiprover work with @just-mitch.

LHerskind committed:
feat: initial multi-proof test

LHerskind commented:
* **#11779** <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/11779?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/AztecProtocol/aztec-packages/11779?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `master`


<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

LHerskind committed:
chore: get rid of proof claims and set the proving cost in the rollup (non-updatable for now)

LHerskind committed:
chore: make proof verification external to reduce codesize

LHerskind committed:
chore: add test for partial proof

just-mitch commented on l1-contracts/test/fees/FeeRollup.t.sol:
This one is hard to follow, but I'm sure you're aware.

LHerskind replied:
Yep. Rewriting the notebook to handle the new proving cost atm so it can be simple again

just-mitch commented on l1-contracts/test/fees/FeeRollup.t.sol:
```diff
@@ -170,18 +178,33 @@ contract FeeRollupTest is FeeModelTestPoints, DecoderBase {
+    uint256 manaBaseFee;
+    {
+      uint256 total = point.outputs.mana_base_fee_components_in_wei.data_cost
+        + point.outputs.mana_base_fee_components_in_wei.gas_cost + 100;
```
I know this is a temporary, but probably good to comment that the 100 is hardcoded into the MinimalFeeModel

just-mitch commented on l1-contracts/test/CheatingRollup.t.sol:
```diff
@@ -0,0 +1,132 @@
+    vm.warp(Timestamp.unwrap(rollup.getTimestampForSlot(Slot.wrap(_slot))));
+  }
+
+  // BRUH, what is this stuff?
```
😬

LHerskind replied:
Ye, wanna get rid of all the cheatcodes.

just-mitch committed:
chore: prover node has no claims or bonds (#11811)

Please read [contributing guidelines](CONTRIBUTING.md) and remove this
line.

---------

Co-authored-by: LHerskind <16536249+LHerskind@users.noreply.github.com>

Maddiaa0 commented on l1-contracts/src/core/RollupCore.sol:
```diff
@@ -321,58 +318,118 @@ contract RollupCore is
+      sr.summedCount += 1;
+
+      // Using the prover id to ensure proof only gets added once
+      require(!sr.hasSubmitted[interim.prover], "go away");
```
"go away"
